### ISSUE TITLE
API Design Systems parser

### ISSUE THREAD

**Issue by char0n** [smizell/api-design-systems#1]
Hi @smizell,

Just wanted to let you know that we're probing the idea of API Design Systems and we've built a semantic parser for it here: https://github.com/swagger-api/apidom/tree/main/packages/apidom-ns-api-design-systems

Cheers!

**Comment by char0n** [comment]
POC of API Design Systems for OpenAPI 3.1 spec validator is available here: https://github.com/swagger-api/apidom/tree/main/packages/apidom-ns-api-design-systems/src/validator/openapi-3-1

**Comment by smizell** [comment]
Wow, I'm sorry to have missed this notification! This is amazing!

**Comment by char0n** [comment]
Will keep you updated - other things we have but are not made public: 
- integration in Microsoft Language Server Procol
- renderer 
- editor integration and instant linting feedback while writing OpenAPI 3.1 spec

**Comment by char0n** [comment]
API Design Systems renderer added to SwaggerEditor@next branch: https://github.com/swagger-api/swagger-editor/pull/3162

**Comment by smizell** [comment]
Wow, amazing. Does this mean it will have official support for the API Design Systems format?

**Comment by char0n** [comment]
Depends on what "official" means ;] 

The whole thing was implemented during Hackathon some time ago, and we're now incorporating it to main branches. The renderer and integration in Microsoft Language Server Protocol is going to be living on main branches and will be part of production build artifacts.

**Comment by smizell** [comment]
That's exciting :) I saw a file that defines SmartBear's guidelines. Was there anything you couldn't express from the guidelines in the API Design System?

**Comment by char0n** [comment]
> That's exciting :) I saw a file that defines SmartBear's guidelines. 

Yes in very limited way. We wanted to limit the scope of implementation so that we could have something workable in short time. The rules are extensive we only defines couple of them in API Design Systems.

> Was there anything you couldn't express from the guidelines in the API Design System?

As far as I remember yes, and we talked about need to extend the API Design Systems specification. We also modeled the file that defines the SmartBear guidelines in a certain way where only `Scenarios` were used, which could be validated with this primitive implementation of validator: https://github.com/swagger-api/apidom/blob/main/packages/apidom-ns-api-design-systems/src/validator/openapi-3-1/validator.ts

**Comment by smizell** [comment]
Let me know what you're missing. I have some future ideas in my head and in notes, but nothing concrete. It would be great to hear about real-world issues.

**Comment by smizell** [comment]
Also, feel free to open a PR to add a link to the editor repo from the API Design System site once support hits the main editor release.

**Comment by char0n** [comment]
The next version of SwaggerEditor now supports the API Design Systems: https://editor-next.swagger.io/

"Edit" -> "Load API Design Systems Fixture"

**Comment by smizell** [comment]
I'm getting a 502 at the moment. I'm looking forward to trying this out!

**Comment by char0n** [comment]
> I'm getting a 502 at the moment. I'm looking forward to trying this out!

Interesting, are you still seeing it now? If yes, then there must something wrong with deployment settings (I'm seeing it OK) and I will let the DevOps team know.

Here is an alternative GitHub Pages deployment: https://ghpages-swagger-editor.swagger.io/

**Comment by char0n** [comment]
At the beginning of this month (August 2022) as part of 2022 Vista Global Hackathon, we've also produced VS Code extension with [SwaggerEditor@next](https://ghpages-swagger-editor.swagger.io/) capabilities. The extension can be found here: https://github.com/swagger-api/apidom-lsp-vscode and support API Design Systems as well.

**Comment by smizell** [comment]
That's so cool! Do you care if I add links to the website for these?

(it's working for me now)

**Comment by char0n** [comment]
> That's so cool! Do you care if I add links to the website for these?

Not at all, it's all public Open Source ;]

> (it's working for me now)

Cool
